src/Symfony/Component/Console/Application.php

```diff
@@ -223,18 +225,36 @@ public function doRun(InputInterface $input, OutputInterface $output)
             // the command name MUST be the first element of the input
             $command = $this->find($name);
         } catch (\Throwable $e) {
-            if (null !== $this->dispatcher) {
-                $event = new ConsoleErrorEvent($input, $output, $e);
-                $this->dispatcher->dispatch(ConsoleEvents::ERROR, $event);
+            if (!($e instanceof CommandNotFoundException && !$e instanceof NamespaceNotFoundException) || 1 !== count($alternatives = $e->getAlternatives()) || !$input->isInteractive()) {
```

Comment:
This proposes to run the alternative even if it's a namespace only, right? A test case would be good to make sure it does not

Reply:
No, if it's a namespace only then the behaviour stays unchanged (Which would giving you an error and giving a list suggested namespaces without the option to run a command)

Reply:
Added another test case to cover namespace error